### PR TITLE
Adding Github Actions workflow 

### LINKED ISSUE #7: CI Pipeline

### PR DESCRIPTION
**Pull request by hanlinc27**
- two yml files (one for Go, one for Node)
- builds and checks linting in the correct directory 
- Closes #48

### CODE REVIEW

File: .github/workflows/node.yml [hunk 1]
```diff
@@ -0,0 +1,35 @@
...
+      working-directory: ./ui
+      run: yarn run build-prod
+    - name: Linting
+      working-directory: ./ui
+      run: |
+        yarn lint:prettier
+        yarn lint:eslint
+    # - name: Testing
+    #   working-directory: ./ui
+    #   run: yarn test
```
Comment (hanlinc27) [inline .github/workflows/node.yml:38]: Will uncomment once there are tests present

File: .github/workflows/node.yml [hunk 2]
```diff
@@ -0,0 +1,35 @@
...
+    - name: Install dependencies
+      working-directory: ./ui
+      run: yarn install
+    - name: Build 
+      working-directory: ./ui
+      run: yarn run build-prod
+    - name: Linting
+      working-directory: ./ui
+      run: |
+        yarn lint:prettier
```
Comment (AbhiPrasad) [inline .github/workflows/node.yml:31]: Let's split this up into two jobs, `Lint Prettier` and `Lint Eslint` so it's easier for us to see what failed

File: .github/workflows/node.yml [hunk 3]
```diff
@@ -0,0 +1,35 @@
...
+        node-version: [10.x, 12.x, 14.x]
+    steps:
+    - uses: actions/checkout@v2
+    - name: Use Node.js ${{ matrix.node-version }}
+      uses: actions/setup-node@v1
+      with:
+        node-version: ${{ matrix.node-version }}
+    
+    - name: Install dependencies
+      working-directory: ./ui
```
Comment (AbhiPrasad) [inline .github/workflows/node.yml:23]: Why `working-directory` on each build over using a `default`?

File: .github/workflows/go.yml [hunk 4]
```diff
@@ -0,0 +1,38 @@
...
+        go-version: ^1.13
+      id: go
+
+    - name: Check out code into the Go module directory
+      uses: actions/checkout@v2
+
+    - name: Get dependencies
+      run: |
+        go get -v -t -d ./...
+        if [ -f Gopkg.toml ]; then
```
Comment (AbhiPrasad) [inline .github/workflows/go.yml]: don't think we need this, we dont use `dep` (https://golang.github.io/dep/)

**Review by AbhiPrasad — CHANGES_REQUESTED** [review]
The Github Actions warnings are kinda annoying, do you think we should comment out the unused variables for now?

Everything looks good, just a quick change about the node workflow.

**Comment by AbhiPrasad** [comment]
Oh yeah, we don't need to test with multiple node versions, `node 12` should be good enough.

File: .github/workflows/node.yml (hunk 3 above)
Comment (hanlinc27) [reply .github/workflows/node.yml:23]: yeah so I know originally I was referencing [this](https://dev.to/shofol/run-your-github-actions-jobs-from-a-specific-directory-1i9e) but it wasn't getting recognized for some reason in github actions so I split it up for [individual jobs](https://github.community/t/use-working-directory-for-entire-job/16747/6)

**Review by AbhiPrasad — APPROVED** [review]

**Review by lesliexin — APPROVED** [review]

**Pull request merged**